### PR TITLE
Numerical stability v40

### PR DESCRIPTION
**Pull request by OrtnerMichael**
# Related Issues

#407 there is no more edgesize setting. when necessary an edgesize is automatically set to np.close( atol=1e-15, rtol=1e-15)

#378 there is no "unit choosing". all computations are now scale invariant !

#376 lib toplevel package now includes all core field functions

Consider development towards a more separated structure
 - core field functions
 - magpylib interface (objects and position/orientation interface)
 - graphics

### CODE REVIEW

**Comment by Alexboiboi** [comment]
I can see you renamed some `cylinder_tile` expression with `cylinder_section`. Is there a good reason not to rename it `cylinder_segment` as in `magnet.CylinderSegment` ?

File: magpylib/_src/fields/special_cel.py [hunk 1]
```diff
@@ -134,36 +134,70 @@ def ellipticPi(x, y):
...
-    C5 = 0.04026699568199808
-    C6 = 0.02642521591631124
-    C7 = 0.01868640268367724
-    C8 = 0.01391747699878044
-    C9 = 0.01076947624905629
-    C10 = 0.008581926385966734
-    return (C2*k2**2 + C3*k2**3 + C4*k2**4 + C5*k2**5
-        + C6*k2**6 + C7*k2**7 + C8*k2**8 + C9*k2**9 + C10*k2**10)
+    errtol = .000001
+    while np.any(abs(g-k) > g*errtol):
```
Comment (Alexboiboi) [inline magpylib/_src/fields/special_cel.py:155]: @OrtnerMichael Are you sure this while loop is safe from being endless?

File: magpylib/_src/input_checks.py [hunk 2]
```diff
@@ -9,6 +9,19 @@
 )
 
 
+def check_field_input(inp, origin):
```
Comment (Alexboiboi) [inline magpylib/_src/input_checks.py:12]: field=True, allowed? Why?

File: magpylib/_src/fields/special_cel.py (hunk 1 above)
Comment (OrtnerMichael) [reply magpylib/_src/fields/special_cel.py:155]: yes - but only Mr. Bulirsch knows the details :)

File: magpylib/_src/input_checks.py (hunk 2 above)
Comment (OrtnerMichael) [reply magpylib/_src/input_checks.py:12]: Doesn't feel right to hand the strings `B` and `H`around library internally. It feels more like a user feature.
This way we get to keep the bool `bh` from level 3 to level 1.

**Comment by OrtnerMichael** [comment]
should now be fine - there is just a merge conflict because of file renamings

**Comment by Alexboiboi** [comment]
> should now be fine - there is just a merge conflict because of file renamings

ready to squash now ;)

**Pull request merged**